src/ZendDiagnostics/Check/RabbitMQ.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace ZendDiagnostics\Check;
+
+use PhpAmqpLib\Connection\AMQPConnection;
+use ZendDiagnostics\Result\Failure;
+use ZendDiagnostics\Result\Success;
+
+/**
+ * Validate that a RabbitMQ service is running
+ */
+class RabbitMQ extends AbstractCheck
+{
+    /**
+     * @var string
+     */
+    protected $host;
+
+    /**
+     * @var integer
+     */
+    protected $port;
+
+    /**
+     * @var string
+     */
+    protected $user;
+
+    /**
+     * @var string
+     */
+    protected $password;
+
+    /**
+     * @var string
+     */
+    protected $vhost;
+
+    /**
+     * @param string  $host
+     * @param integer $port
+     * @param string  $user
+     * @param string  $password
+     * @param string  $vhost
+     */
+    public function __construct(
+        $host = 'localhost',
+        $port = 5672,
+        $user = 'guest',
+        $password = 'guest',
+        $vhost = '/'
+    ) {
+        $this->host     = $host;
+        $this->port     = $port;
+        $this->user     = $user;
+        $this->password = $password;
+        $this->vhost    = $vhost;
+    }
+
+    /**
+     * Perform the check
+     *
+     * @see \ZendDiagnostics\Check\CheckInterface::check()
+     * @return Failure|Success
+     */
+    public function check()
+    {
+        if (!class_exists('PhpAmqpLib\Connection\AMQPConnection')) {
+            return new Failure('PhpAmqpLib is not installed');
+        }
+
+        $conn = new AMQPConnection(
```

Comment:
Exactly :-) Runner is responsible for catching that ... you can't be sure what kind of exception that is. Consider running diag in an app stack, where you actually catch some errors, but you let others pass through and i.e. return JSON with `{success:false}` or something